platform/darwin/src/MGLStyle.mm

```diff
@@ -572,6 +572,20 @@ - (MGLImage *)imageForName:(NSString *)name
 
 #pragma mark Style transitions
 
+- (void)setTransition:(MGLTransition)transition
+{
+    [self setTransitionDuration:transition.duration];
+    [self setTransitionDelay:transition.delay];
```

Comment:
We should probably get rid of the separate internal `-setTransitionDuration:` and `-setTransitionDelay:` since externally we only set full transition structures containing both values.